stakingModel/test/GoodFundManager.e2e.test.js

```diff
@@ -0,0 +1,117 @@
+const NETWORK = "test";
+
+async function proposeAndRegister(addr, registrar, proposalId, absoluteVote, avatarAddress, fnd) {
+  const transaction = await registrar.proposeScheme(avatarAddress, addr, NULL_HASH, "0x00000010", NULL_HASH);
+  proposalId = transaction.logs[0].args._proposalId;
+  const voteResult = await absoluteVote.vote(proposalId, 1, 0, fnd);
+  return voteResult.logs.some(e => e.event === 'ExecuteProposal');
+}
+
+async function next_interval() {
+  let blocks = 5760;
+  for (let i = 0; i < blocks; ++i)
+    await web3.currentProvider.send(
+      { jsonrpc: "2.0", method: "evm_mine", id: 123 },
+      () => {}
+    );
+}
+
+contract("GoodCDaiReserve - network e2e tests", ([founder, staker]) => {
+  let dai, cDAI, simpleStaking, goodReserve, goodFundManager, goodDollar, marketMaker, contribution;
+  let deploy_settings, ubiBridgeRecipient, avatarAddress, registrar, absoluteVote, proposalId, setReserve;
+
+  before(async function() {
+    const staking_file = await fse.readFile("releases/deployment.json", "utf8");
+    const dao_file = await fse.readFile("../releases/deployment.json", "utf8");
+    const staking_deployment = await JSON.parse(staking_file);
+    const dao_deployment = await JSON.parse(dao_file);
+    const staking_addresses = staking_deployment[NETWORK];
+    const dao_addresses = dao_deployment[NETWORK];
+    avatarAddress = dao_addresses.Avatar;
+    ubiBridgeRecipient = staking_addresses.UBIScheme;
+    dai = await DAIMock.at(staking_addresses.DAI);
+    cDAI = await cDAIMock.at(staking_addresses.cDAI);
+    simpleStaking = await SimpleDAIStaking.at(staking_addresses.DAIStaking);
+    goodReserve = await GoodReserve.at(staking_addresses.Reserve);
+    goodFundManager = await GoodFundsManager.at(staking_addresses.FundManager);
+    marketMaker = await MarketMaker.at(staking_addresses.MarketMaker);
+    contribution = await ContributionCalculation.at(staking_addresses.Contribution);
+    goodDollar = await GoodDollar.at(dao_addresses.GoodDollar);
+    registrar = await SchemeRegistrar.at(dao_addresses.SchemeRegistrar);
+    absoluteVote = await AbsoluteVote.at(dao_addresses.AbsoluteVote);
+    deploy_settings = await fse.readFile("../migrations/deploy-settings.json", "utf8");
+    console.log(deploy_settings);
+    // schemes
+    setReserve = await FundManagerSetReserve.new(avatarAddress, goodFundManager.address, goodReserve.address);
+  });
+
+  it("should mints cdai and updates exchange rate", async () => {
+    await dai.mint(staker, web3.utils.toWei("100", "ether"));
+    await dai.approve(simpleStaking.address, web3.utils.toWei("100", "ether"), {
+      from: staker
+    });
+    await simpleStaking
+    .stakeDAI(web3.utils.toWei("100", "ether"), {
+      from: staker
+    })
+    .catch(console.log);
+    await cDAI.exchangeRateCurrent();
+    let rate = await cDAI.exchangeRateStored();
+    expect(rate.toString()).to.be.equal("10201010101010101010101010101");
```

Comment:
1. you are testing the functionality of the cDAI mock here. we dont test mocks. you test what you expect to happen from our contracts.
2. hard coded number without explanation

Reply:
it's an irrelevant test, removed from the suggested version
db43a7633912fed7aa89a7ff580fbe5238d650b5